go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -2,13 +2,32 @@
 
 require "dependabot/shared_helpers"
 require "dependabot/errors"
+require "dependabot/go_modules/file_parser"
 require "dependabot/go_modules/file_updater"
 require "dependabot/go_modules/native_helpers"
 
 module Dependabot
   module GoModules
     class FileUpdater
       class GoModUpdater
+        # Turn off the module proxy for now, as it's causing issues with
+        # private git dependencies
+        ENVIRONMENT = { "GOPRIVATE" => "*" }.freeze
+
+        RESOLVABILITY_ERROR_REGEXES = [
+          /go: .*: git fetch .*: exit status 128/.freeze,
+          /verifying .*: checksum mismatch/.freeze,
+          /build .*: cannot find module providing package/.freeze
+        ].freeze
+
+        MODULE_PATH_MISMATCH_REGEXES = [
+          /go: ([^@\s]+)(?:@[^\s]+)?: .* has non-.* module path "(.*)" at/,
+          /go: ([^@\s]+)(?:@[^\s]+)?: .* unexpected module path "(.*)"/,
+          /go: ([^@\s]+)(?:@[^\s]+)?: .* declares its path as: ([\S]*)/m
+        ].freeze
+
+        attr_reader :dependencies, :go_mod, :go_sum, :credentials
```

Comment:
This was at the bottom of the class. I've moved it here as it typically lives at the top of the class declaration.

Reply:
Previously it was private and now it's public, if we don't need to access it from outside I'd typically declare it right after the `private` declaration, but not super bothered.

Reply:
Good catch - fixed!